src/display_context.cpp

```diff
@@ -87,21 +87,59 @@ display_context::can_move_result display_context::unit_can_move(const unit& u) c
 	const team& current_team = get_team(u.side());
 
 	can_move_result result = {false, false};
-	for(const map_location& adj : get_adjacent_tiles(u.get_location())) {
-		if (map().on_board(adj)) {
-			if(!result.attack_here) {
-				const unit_map::const_iterator i = units().find(adj);
-				if (i.valid() && !i->incapacitated() && current_team.is_enemy(i->side())) {
-					result.attack_here = true;
+
+	// Generating pairs of min-max ranges of all units weapons
+	const auto& attacks = u.attacks();
+	std::vector<std::pair<int, int>> distance_ranges;
+	for (const auto& attack : attacks) {
+		int min_range = attack.min_range();
+		int max_range = attack.max_range();
+		distance_ranges.emplace_back(min_range , max_range);
```

Comment:
I'm wondering if this should be skipping attacks that have a `[disable]` special active.

Reply:
I don't think so, you'll have to specifically check for the disable special and exclude it.

Reply:
Unit preview pane uses `u.attacks()` and it still shows disabled attacks.

Reply:
@hladkyjakub can you please test it on units with a `[disable]` special?

Reply:
> I don't think so, you'll have to specifically check for the disable special and exclude it.

That's what I'm saying. Should it?

Reply:
Right, it's really not necessary to check for attacks that are disabled.

Reply:
So you're saying that the orb should be green/yellow if you have a range 2 attack that's disabled and there's an enemy 2 hexes away?

Reply:
Currently even AI think it can attack with disabled weapons/ weapons that loose all strikes due to a special with condition.
Same goes for Lua btw.
...
So maybe theres a reason for that? (Idk)

Reply:
So...
What should be done about this?

Reply:
Sounds like it's not something to deal with in this PR.

Reply:
... that looks like the last objection, I've got on this PR.